peermanager/peermanager.go

```diff
@@ -192,19 +99,29 @@ func (pm *PeerManager) stopPeerHandler(p peer.ID) {
 	delete(pm.peerQueues, p)
 }
 
-func (pm *PeerManager) sendMessage(ms *sendPeerMessage) {
-	if len(ms.targets) == 0 {
+// SendMessage is called to send a message to all or some peers in the pool;
+// if targets is nil, it sends to all.
+func (pm *PeerManager) SendMessage(initialEntries []*wantlist.Entry, entries []*bsmsg.Entry, targets []peer.ID, from uint64) {
```

Comment:
This is rather leaky. What if we just waited for the `initialEntries` to be propegated along with the `Connected` message? We'll likely send two packets but that's not _that_ bad (and we can try buffering elsewhere if necessary.

This would also allow us to revert the signature change to `Startup`. `Connected` would start then send the initial entries in two steps.

Reply:
see latest code -- its a bit tricky cause of the nature of the wantlist itself -- the fact that it tracks sessions.

But I think I have a solution that avoids sending entries with SendMessage